### ISSUE TITLE
Virtual Schema Error

### ISSUE THREAD

**Issue by ma-blaetke** [sebriois/biomart#18]
hi,

I am trying to retrieve transcript information from "_http://plants.ensembl.org/biomart"_ Server using the dataset _"athaliana_eg_gene"._ Unfortunanetly its not working and I get an error:

**Query ERROR: caught BioMart::Exception::Usage: WITHIN Virtual Schema : default, Dataset athaliana_eg_gene NOT FOUND**

I found out the the virtual schema is somehow set to _'default'_, but in my case I need another one here named _'plants_mart'_. Is there a way to set this option on my own? Or what solution do you suggest?

best regards

mary

**Comment by llrs** [comment]
Hi Marie, 
you can view the available dataset with `server.show_datasets()`, have a look at the [Readme](https://github.com/sebriois/biomart#usage), and then change to any other with `plants_mart = server.datasets['plants_mart']`

HTH

**Comment by ma-blaetke** [comment]
Thank you for your answer, but I think there is a missunderstanding.

Again, I connect to the server _http://plants.ensembl.org/biomart:_
 `server = BiomartServer( "http://plants.ensembl.org/biomart" )` 
From there I choose the dataset _athaliana_eg_gene_:
`ath = server.datasets['athaliana_eg_gene']`
The `server.show_datasets()` list a lot of data sets, e.g. the one I want to use, but it doesn't list 'plants_mart'.
From the Error-Message **Query ERROR: caught BioMart::Exception::Usage: WITHIN Virtual Schema : default, Dataset athaliana_eg_gene NOT FOUND** I got that something is wrong with the virtual schema. As the error message says and the code of this biomart 0.9.0 module shows, the virtual schema is set to `default`. And that is the problem. In my case, I need to change, the virtual schema that is applied. I did some research and found out that for my purpose It has to be set to "plant_mart" and not "default". 
The module should offer the possibility to change the virtual schema if neccessary, e.g. by allowing the user to set another parameter by calling the search method. 
In the Readme I found no option to change the virtual schema.

**Comment by llrs** [comment]
The data set is listed correctly, but the schema applied is not correct? You can provide the schema when creating the class server: [See the source](https://github.com/sebriois/biomart/blob/050a41a860e5ce08a4198f7370a219cf10e31db5/biomart/database.py#L4-L21).
You can provide the following attributes to a server, the server (obviously), the name, the display name, the virtual_schema, and verbose type (true or false).

If you use `server = BiomartServer( "http://plants.ensembl.org/biomart" , virtual_schema = "plant_mart")`, you should be able to connect correctly to that dataset.

**Comment by sebriois** [comment]
Hi Luis and Marie,

Luis answer is the way to achieve what you want to do Marie, but the sad part is that the [code](https://github.com/sebriois/biomart/blob/master/biomart/dataset.py#L219) still uses the 'default' virtualSchema, although you pass on a value for it when creating the BiomartServer instance.

So, I'm preparing a fix for this issue. I'll let you know once it's pushed and available for a `pip install --upgrade biomart`

Seb

**Comment by sebriois** [comment]
Hi Mary,

alright - fixed and available in v0.9.2

The virtualSchema will be automatically guessed for you, so you don't have to specify it.

So, here is an example, searching for the arabidopsis gene AT1G14920.
It is using the default attributes here in the output (ensembl_transcript_id and ensembl_gene_id) but you can specify the ones you want in the `query` dict (see below)

```bash
Python 3.4.3 (v3.4.3:9b73f1c3e601, Feb 23 2015, 02:52:03) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from biomart import BiomartServer
>>> server = BiomartServer( "http://plants.ensembl.org/biomart" )
>>> ath = server.datasets['athaliana_eg_gene']
>>> query = {
...   'filters': {
...     'ensembl_gene_id': 'AT1G14920'
...   }
... } 
>>> response = ath.search(query)
>>> for line in response.iter_lines():
...   print(line.decode())
... 
AT1G14920.1	AT1G14920
>>> 
```

Best,
Seb
